src/bots/filler.ts

```diff
@@ -666,6 +690,17 @@ export class FillerBot implements Bot {
 	}
 
 	protected async confirmPendingTxSigs() {
+		// Check hasEnoughSolToFill before confirm the txs to not jam up the event loop if we're trying to rebalance
+		const release = await this.hasEnoughSolToFillMutex.acquire();
+		try {
+			if (!this.hasEnoughSolToFill) {
+				logger.info(`Not enough SOL to fill, skipping confirm pending tx sigs`);
+				return;
+			}
+		} finally {
+			release();
+		}
```

Comment:
you shouldn't need to block this task while rebalancing, since it'll be chipping away at `this.pendingTxSigsToconfirm` and essentially be a noop when that's empty